dashboard/test/models/user_test.rb

```diff
@@ -13,6 +12,36 @@ class UserTest < ActiveSupport::TestCase
     assert_equal '21+', teacher.age
   end
 
+  # Disable this test if and when we do require teachers to complete school data
+  test 'school info should not be validated' do
+    school_attributes = {
+      country: 'US',
+      school_type: SchoolInfo::SCHOOL_TYPE_PUBLIC,
+      state: nil
+    }
+    teacher = create :teacher, school_info_attributes: school_attributes
+    teacher.save!
+    assert teacher.valid?, teacher.errors.full_messages
+  end
+
+  test 'identical school info should not be duplicated in the database' do
+    school_attributes = {
+      country: 'US',
+      school_type: SchoolInfo::SCHOOL_TYPE_PUBLIC,
+      state: 'CA'
+    }
+    teacher = create :teacher, school_info_attributes: school_attributes
+    teacher.save!
+
+    teacher2 = create :teacher, school_info_attributes: school_attributes
+    teacher2.save!
```

Comment:
Not quite, as FactoryGirl `create` doesn't throw. If the code were `save` it would be redundant, though `save!` is not completely redundant (from a functional standpoint).